core/templates/dev/head/components/forms/custom-forms-directives/html-select.directive.html

```diff
@@ -16,6 +16,13 @@
 </div>
 
 <style>
+  .oppia-html-select > ul > li > a {
+    display: block;
+    padding: 3px 20px;
+  }
+  .oppia-html-select > ul > li > a:hover {
```

Comment:
Is it similar to the CSS on line 58? (Do you know why this issue happened in the first place, although it was working fine earlier?)

Reply:
@vojtechjelinek are you familiar with how `third_party.css` is generated? I'm not sure where to check for changes that might have introduced this bug since I think `third_party.css` is generated by the `scripts.start` script (so no `git blame`)

Reply:
@U8NWXD, I think this issue might have introduced because of some upgrade in third_party library,  third_party.css file is a minified version of all the third_party style sheets. I think you can find the comment which has information about the third_party which is introducing the given style to the class/id. (similar to the image below)

![image](https://user-images.githubusercontent.com/16653571/66154618-888e5900-e63b-11e9-941f-e5678417740e.png)

This info will help you find the cause of the issue (maybe).

Reply:
This bug appears to have been introduced by the migration to Bootstrap 4.3.1 from 3.3.4 (ba55ea9b5df8d6c8d3d2b82ca0c11800f8409d81), which was not included in the [2.8.5 release](https://github.com/oppia/oppia/commits/release-2.8.5). Bootstrap 4 now uses syntax like this for dropdowns:

```html
<div class="dropdown">
  <button class="btn btn-secondary dropdown-toggle" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
    Dropdown button
  </button>
  <div class="dropdown-menu" aria-labelledby="dropdownMenuButton">
    <a class="dropdown-item" href="#">Action</a>
    <a class="dropdown-item" href="#">Another action</a>
    <a class="dropdown-item" href="#">Something else here</a>
  </div>
</div>
```
(see [bootstrap docs](https://getbootstrap.com/docs/4.3/components/dropdowns/)).

We use Angular UI Bootstrap, which is [only compatible with Bootstrap 3.x](https://angular-ui.github.io/bootstrap/). The dropdown menu is created by `core/templates/dev/head/components/forms/custom-forms-directives/html-select.directive.html`, which includes this code:

```html
  <ul uib-dropdown-menu style="max-width: 80%; overflow: auto; width: auto;">
    <li ng-repeat="choice in options" style="float: left;">
      <a ng-click="select(choice.id)">
        <angular-html-bind class="oppia-html-select-option protractor-test-html-select-option" html-data="choice.val">
        </angular-html-bind>
      </a>
    </li>
  </ul>
```

`uib-dropdown-menu` is an Angular UI Bootstrap directive that generates code like this:

```html
  <ul uib-dropdown-menu="" style="max-width: 80%; overflow: auto; width: auto; position: absolute; will-change: transform; top: 0px; left: 0px; transform: translate3d(0px, 30px, 0px);" class="dropdown-menu show" x-placement="bottom-start">
    <li ng-repeat="choice in options" style="float: left;" class="">
      <a ng-click="select(choice.id)">
        <angular-html-bind class="oppia-html-select-option protractor-test-html-select-option" html-data="choice.val"><p>pass</p></angular-html-bind>
      </a>
    </li>
    <li ng-repeat="choice in options" style="float: left;" class="">
      <a ng-click="select(choice.id)">
        <angular-html-bind class="oppia-html-select-option protractor-test-html-select-option" html-data="choice.val"><p>fail</p></angular-html-bind>
      </a>
    </li>
  </ul>
```

If we add `dropdown-item` as a class for the `<li>` elements like this:

```html
  <ul uib-dropdown-menu style="max-width: 80%; overflow: auto; width: auto;">
    <li ng-repeat="choice in options" style="float: left;" class="dropdown-item">
      <a ng-click="select(choice.id)">
        <angular-html-bind class="oppia-html-select-option protractor-test-html-select-option" html-data="choice.val">
        </angular-html-bind>
      </a>
    </li>
  </ul>
```

we get nice dropdown options that, while not identical to the production version, look nice:

![Screen Shot 2019-10-08 at 09 13 04](https://user-images.githubusercontent.com/19878639/66413238-dc869e00-e9ab-11e9-831d-c35503e8f43f.png)

I'll push a commit adding the `dropdown-item` class in a few hours. Using Angular UI Bootstrap with Bootstrap 4 might cause future issues though.

Reply:
@DubeySandeep the lint tests timed out, but they pass when I run them locally. PTAL

Reply:
I have restarted the tests once. 

I am not sure why the tests time out though. @kevinlee12 any idea of why this is happening?